src/Conversion/KrnlToLLVM/KrnlInstrument.cpp

```diff
@@ -59,8 +59,8 @@ class KrnlInstrumentOpLowering : public ConversionPattern {
     Value tag = create.llvm.constant(
         IntegerType::get(context, 64), (int64_t)instrumentOp.getTag());
     StringRef nodeName;
-    if (instrumentOp.getNodeName().has_value())
-      nodeName = instrumentOp.getNodeName().value();
+    if (auto nameLoc = loc.dyn_cast<NameLoc>())
+      nodeName = nameLoc.getName();
```

Comment:
Could we try to get from both NodeName and NameLoc? Is there any case that we have NodeName but not NameLoc and vice versa?

Reply:
Node name in onnx model is set in both `onnx_node_name`attribute (NodeName) and location (NameLoc) when importing the model. The Nodename is often lost by rewriting, but the NameLoc is not lost. So, there are the case that it has NameLoc, but not NodeName.  I think it is better to use NameLoc here.

Reply:
The node_name attribute is not the predefined attribute for ONNX Ops, and will be lost by our rewriting.

Reply:
I see. I will add the cases. Thanks!

Reply:
I added the case for FusedLoc and FileLineColLoc. In the FusedLoc case, name of each location are combined and the combined name is used as nodeName. In the FileLineColLoc,  filename and fileline are combined and the combined name is used.